src/main/java/com/maxmind/db/Decoder.java

```diff
@@ -350,14 +350,16 @@ private static boolean decodeBoolean(int size)
         return constructor.newInstance(parameters);
     }
 
-    private static <T> Constructor findConstructor(Class<T> cls)
+    private static <T> Constructor<T> findConstructor(Class<T> cls)
         throws ConstructorNotFoundException {
         Constructor<?>[] constructors = cls.getConstructors();
         for (Constructor<?> constructor : constructors) {
             if (constructor.getAnnotation(MaxMindDbConstructor.class) == null) {
                 continue;
             }
-            return constructor;
+            @SuppressWarnings("unchecked")
+            Constructor<T> constructor2 = (Constructor<T>) constructor;
+            return constructor2;
```

Comment:
Any reason why this can't just be `return (Constructor<T>) constructor;`?

Reply:
I couldn't add a `@SuppressWarnings` when written that way. Apparently the grammar doesn't allow it!